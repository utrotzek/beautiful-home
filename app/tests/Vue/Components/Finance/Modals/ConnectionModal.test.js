import {mount} from "@vue/test-utils";
import connectionModals from "../../../../../resources/js/components/Finance/Modals/ConnectionModal";
import Vue from "vue";

import PlanningDataFixtures from "../Fixtures/PlanningFixtures";
import AccountingFixtures from "../Fixtures/AccountingFixtures";

describe("accounting and planning connection modal", () => {
    let wrapper = null;
    let planningData = PlanningDataFixtures[0];
    let accountingData = AccountingFixtures[0];
    accountingData.remainingAmount = -100;

    beforeEach(() => {
        wrapper = mount(connectionModals,{
            propsData: {
                accounting: accountingData,
                planning: planningData,
                show: true,
                desiredAmount: -100
            }
        });
    });

    it ("renders properly", () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it ("can fire close event", async () =>{
        wrapper.find(".close").trigger("click");
        await Vue.nextTick();
        expect(wrapper.emitted().close).toBeTruthy();
    });

    it ("can fire do connection event", async () =>{
        wrapper.vm.localDesiredAmount = -50;
        wrapper.find(".do-connection").trigger("click");
        await Vue.nextTick();

        expect(wrapper.emitted().doAccountingPlanningConnection[0]).toEqual([
            planningData,
            accountingData,
            -50
        ]);
    });

    it ("desired amount can be set to full remaining amount", async () =>{
        wrapper.vm.localDesiredAmount = -50;
        wrapper.find(".use-full-amount").trigger("click");
        await Vue.nextTick();

        expect(wrapper.vm.localDesiredAmount).toEqual(-100);
    });

    it ("local desired amount will be in sync with desired amount", async () =>{
        wrapper.setProps({desiredAmount: 100});
        await Vue.nextTick();
        expect(wrapper.vm.localDesiredAmount).toEqual(100);

        wrapper.setProps({desiredAmount: -120});
        await Vue.nextTick();
        expect(wrapper.vm.localDesiredAmount).toEqual(-120);

    });
});
