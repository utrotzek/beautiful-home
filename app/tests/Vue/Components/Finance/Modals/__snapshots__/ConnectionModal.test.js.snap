// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accounting and planning connection modal renders properly 1`] = `
<div
  class="modal-mask"
>
  <div
    class="modal-wrapper"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
          >
            
                            Zuordnen
                        
          </h5>
           
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
         
        <div
          class="modal-body"
        >
          <p>
            <strong>
              Rewe
            </strong>
             mit Kostenstelle 
            <strong>
              Einkaufen
            </strong>
             verknüpfen.
          </p>
           
          <div
            class="form-group row"
          >
            <label
              class="col-2 col-form-label"
              for="desiredAmount"
            >
              Betrag
            </label>
             
            <div
              class="col-6"
            >
              <input
                class="form-control"
                id="desiredAmount"
                type="text"
              />
            </div>
             
            <div
              class="col-4"
            >
              <button
                class="btn btn-secondary use-full-amount"
              >
                <span
                  class="fa fa-hand-holding-usd"
                />
              </button>
            </div>
          </div>
           
          <div>
            <div>
              
                                Es bleibt dann ein Restbetrag von 0.00 €
                            
            </div>
          </div>
        </div>
         
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-secondary"
            type="button"
          >
            <i
              class="fas fa-ban"
            />
            
                            Abbrechen
                        
          </button>
           
          <button
            class="btn btn-primary do-connection"
            type="button"
          >
            <i
              class="fas fa-link"
            />
            
                            Zuordnen
                        
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
